### PR TITLE
WIP: Web3 integration

### PR DESCRIPTION
**Pull request by DeV1doR**
Net namespace:
- net_listening;
- net_peerCount;
- net_version;

Web3 namespace:
- web3_clientVersion;
- web3_sha3;

Eth namespace:
- eth_protocolVersion;
- eth_syncing;
- eth_coinbase;
- eth_mining;
- eth_hashrate;
- eth_gasPrice;
- eth_accounts;
- eth_getBalance;
- eth_getStorageAt;
- eth_getCode;
- eth_call;
- eth_estimateGas;
- eth_getBlockByHash;
- eth_getBlockByNumber;
- eth_getUncleByBlockHashAndIndex;
- eth_getUncleByBlockNumberAndIndex;
- eth_getBlockTransactionCountByHash;
- eth_getBlockTransactionCountByNumber;
- eth_getUncleCountByBlockHash;
- eth_getUncleCountByBlockNumber; 
- eth_getLogs;
- eth_getTransactionCount;
- eth_getTransactionByHash;
- eth_getTransactionByBlockHashAndIndex;
- eth_getTransactionByBlockNumberAndIndex;
- eth_getTransactionReceipt;

### CODE REVIEW

File: web3/net/service.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+// PublicNetAPI is the eth_ prefixed set of APIs in the Web3 JSON-RPC spec.
+type PublicNetAPI struct {
+	networkVersion string
+	peerCount      hexutil.Big
+}
+
+// NewService creates an instance of the public Net Web3 API.
+func NewService(clientCtx *service.Context) *PublicNetAPI {
+	peerCount := big.NewInt(int64(len(clientCtx.Cfg.P2P.PersistentPeers)))
```
Comment (DeV1doR) [inline web3/net/service.go:18]: This is not true, it does not represents the current connected peers. 
https://docs.ethermint.zone/basics/json_rpc.html#json-rpc-methods

I am not sure if we need Listening and PeerCount at all, did u check if another web3 rpcs like Matic/Binance provide this info?

File: web3/net/service.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	// TODO: how to fetch if the node is open to network connection??
+	return true
+}
+
+func (svc *PublicNetAPI) PeerCount() hexutil.Big {
+	return svc.peerCount
+}
+
+func (svc *PublicNetAPI) Version() string {
+	return fmt.Sprintf("%s", svc.networkVersion)
```
Comment (DeV1doR) [inline web3/net/service.go:35]: HashToBigInt

Comment (DeV1doR) [reply web3/net/service.go:35]: Sence of HashToBigInt in general - for the smart contracts, chainId is int there, it is only in the tendermint it is a string.

File: web3/net/service.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package net
+
+import (
+	"fmt"
+	"github.com/Oneledger/protocol/service"
```
Comment (DeV1doR) [inline web3/net/service.go:5]: circular import?

**Review by alexjlan — APPROVED** [review]

**Review by alexjlan — APPROVED** [review]

File: data/governance/store.go [hunk 4]
```diff
@@ -106,13 +106,13 @@ func (st *Store) Get(key string, optKey string) ([]byte, error) {
 		panic(errors.Wrap(err, "Unable to get Last Update Height"))
 	}
 	// Get the Options from the last update Height
-	versionedKey := storage.StoreKey(fmt.Sprint(luh) + storage.DB_PREFIX + key)
+	versionedKey := storage.StoreKey(string(luh) + storage.DB_PREFIX + key)
```
Comment (DeV1doR) [inline data/governance/store.go:109]: Fix from london update

**Pull request closed without merging**